arm_robots/scripts/cartesian_teleop.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+import argparse
+import sys
+from threading import Thread
+from typing import List
+
+import numpy as np
+from inputs import get_gamepad
+
+import ros_numpy
+import rospy
+from arc_utilities import ros_init
+from arm_robots.get_robot import get_moveit_robot
+from arm_robots.logitech import Logitech
+from tf.transformations import quaternion_from_euler
+
+
+class CartesianTeleop:
+    """
+    use the roslaunch file to start this and the joy node. Then use the D-pad and left joystick to move in XYZ
+    The "x" button (on logitech controller) will switch which tool name is being controller
+
+    """
+    def __init__(self, robot_name: str, group_name: str, tool_names: List[str]):
+        self.gamepad = Logitech()
+        self.group_name = group_name
+        self.tool_names = tool_names
+
+        self.robot = get_moveit_robot(robot_name)
+        self.robot.connect()
+
+        self.robot.store_current_tool_orientations(self.tool_names)
+
+        self.current_tool_idx = 0
+        self.delta_scale_factor = 0.01
+
+        self.should_disconnect = False
+        self.buttons_thread = Thread(target=self.buttons_thread_func)
+        self.buttons_thread.start()
+
+        r = rospy.Rate(10)
+        while not rospy.is_shutdown():
+            self.send_commands()
+            r.sleep()
+
+        self.should_disconnect = True
+
+    def get_current_positions(self):
+        target_positions = []
+        for tool_name in self.tool_names:
+            current_tool_pose = self.robot.get_link_pose(tool_name)
+            current_tool_position = ros_numpy.numpify(current_tool_pose.position)
+            target_positions.append(current_tool_position)
+        return target_positions
+
+    def buttons_thread_func(self):
+        while True:
+            events = get_gamepad()
+            for event in events:
+                if self.gamepad.x_clicked(event):
+                    if self.current_tool_idx == len(self.tool_names) - 1:
```

Comment:
Personal preference, but avoid `if else` as much as possible
```
self.current_tool_idx = (self.current_tool_idx + 1) % len(self.tool_names)
```

or

```
self.current_tool_idx += 1
if self.current_tool_idx == len(self.tool_names):
  self.current_tool_idx = 0
```
```

Reply:
I like the later